reactive/kubernetes_worker.py

```diff
@@ -678,6 +678,10 @@ def configure_kubelet(dns, ingress_ip):
         kubelet_opts['cloud-provider'] = 'azure'
         kubelet_opts['cloud-config'] = str(kubelet_cloud_config_path)
         kubelet_opts['provider-id'] = azure.vm_id
+    else:
+        kubelet_opts['cloud-provider'] = ''
+        kubelet_opts['cloud-config'] = ''
+        kubelet_opts['provider-id'] = ''
```

Comment:
This shouldn't be necessary. When these args are not explicitly set, they will not be in kubelet_opts, so they will get automatically removed when [configure_kubernetes_service](https://github.com/charmed-kubernetes/layer-kubernetes-common/blob/cda05cfceb864f7488483e791470473db85cca36/lib/charms/layer/kubernetes_common.py#L369-L371) is called (if they were previously set).